src/inc/corinfo.h

```diff
@@ -2335,9 +2344,9 @@ class ICorStaticInfo
     // Quick check whether the type is a value class. Returns the same value as getClassAttribs(cls) & CORINFO_FLG_VALUECLASS, except faster.
     virtual BOOL isValueClass(CORINFO_CLASS_HANDLE cls) = 0;
 
-    // If this method returns true, JIT will do optimization to inline the check for
+    // Decides how the JIT should do the optimization to inline the check for
     //     GetTypeFromHandle(handle) == obj.GetType()
-    virtual BOOL canInlineTypeCheckWithObjectVTable(CORINFO_CLASS_HANDLE cls) = 0;
+    virtual CorInfoObjectVTableTypeCheckInliningResult canInlineTypeCheckWithObjectVTable(CORINFO_CLASS_HANDLE cls) = 0;
```

Comment:
Modifying existing JIT/EE interface API may require more changes in desktop. (I believe we are still testing the current JIT with desktop internally.) @AndyAyersMS Is it acceptable?

Reply:
We are still mirroring things over and testing all the desktop jits. So changing an existing API is not straightforward as you'd also have to update jit32 and jit64.

Adding a new call is somewhat easier as the other jits won't use it and so don't need to be touched.